### PR TITLE
LR scheduler docs update

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/5675
Tries to clarify the different ways that lr schedulers can be defined in `configure_optimizers` as some users find it confusing how to define a scheduler that steps after each batch.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/common/optimizers.rst [hunk 1]
```diff
@@ -130,7 +124,21 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
       scheduler2 = LambdaLR(optimizer2, ...)
       return [optimizer1, optimizer2], [scheduler1, scheduler2]
 
-   # Alternatively
+Schedulers which ``.step()`` method are conditioned on metric values (for example the ``ReduceLROnPlateau`` 
+scheduler) enforces that the output from ``configure_optimizers`` should be returned as dicts with the 
+keyword ``monitor`` defined as the metric that the scheduler should be conditioned on.
```
Comment (rohitgr7) [inline docs/source/common/optimizers.rst:129]:
```suggestion
Schedulers in which ``.step()`` method is conditioned on metric values (for example the
:class:`~torch.optim.lr_scheduler.ReduceLROnPlateau` scheduler) enforces that the output from ``configure_optimizers`` should be returned as dicts with the  keyword ``monitor`` defined as the
metric that the scheduler should be conditioned on.
```

File: docs/source/common/optimizers.rst [hunk 2]
```diff
@@ -141,15 +149,33 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
           {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
       )
 
-   # Same as above with additional params passed to the first scheduler
+As default all schedulers will be called after each epoch ends. To change this behaviour, each scheduler should be
```
Comment (rohitgr7) [inline docs/source/common/optimizers.rst:152]:
```suggestion
By default, all schedulers will be called after each epoch ends. To change this behaviour, each scheduler should be
```

File: docs/source/common/optimizers.rst [hunk 3]
```diff
@@ -141,15 +149,33 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
...
+returned as a dict which can contain the following keywords:
+
+* ``scheduler`` (required): the actual scheduler class
+* ``monitor`` (optional): same as above, defines the value that certain schdulers should be conditioned on
```
Comment (rohitgr7) [inline docs/source/common/optimizers.rst:156]:
```suggestion
* ``monitor`` (optional): same as above, defines the value that certain schedulers should be conditioned on
```

File: docs/source/common/optimizers.rst [hunk 4]
```diff
@@ -141,15 +149,33 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
...
+* ``interval`` (optional): either ``epoch`` (default) for stepping after each epoch ends or ``step`` for stepping
+  after each optimization step
+* ``frequency`` (optional): how many epochs/steps should pass between calls to ``scheduler.step()``. Default is 1,
+  corresponding to updating the learning rate after every epoch/step.
+* ``strict`` (optional): if set to ``True`` will enforce that value specified in ``monitor`` is available while trying
+  to call ``scheduler.step()``, and stop training if not found. If ``False`` will only give an warning and continue training
```
Comment (rohitgr7) [inline docs/source/common/optimizers.rst:162]:
```suggestion
  to call ``scheduler.step()``, and stop training if not found. If ``False`` will only give a warning and continue training
```

File: docs/source/common/optimizers.rst [hunk 5]
```diff
@@ -101,6 +101,8 @@ In the case of multiple optimizers, Lightning does the following:
 Learning rate scheduling
 ------------------------
 Every optimizer you use can be paired with any `LearningRateScheduler <https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate>`_.
+In the basic usecase, the scheduler (or multiple schedulers) should be returned as the second output from the ``.configure_optimizers``
+method:
```
Comment (rohitgr7) [inline docs/source/common/optimizers.rst:105]:
```suggestion
In the basic use-case, the scheduler (or multiple schedulers) should be returned as the second output from the ``.configure_optimizers``
method:
```

**Review by rohitgr7 — APPROVED** [review]
nice!

File: docs/source/common/optimizers.rst [hunk 6]
```diff
@@ -141,15 +149,33 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
...
+* ``strict`` (optional): if set to ``True`` will enforce that value specified in ``monitor`` is available while trying
+  to call ``scheduler.step()``, and stop training if not found. If ``False`` will only give a warning and continue training
+  (without calling the scheduler).
+* ``name`` (optional): if using the ``LearningRateMonitor`` callback to monitor the learning rate progress, this
+  keyword can be used to specify a specific name.
+
+.. testcode::
+
+   # Same as the above example with additional params passed to the first scheduler
+   # In this case the ReduceLROnPlateau will step after every 10 processed batch
```
Comment (awaelchli) [inline docs/source/common/optimizers.rst:170]:
```suggestion
   # In this case the ReduceLROnPlateau will step after every 10 processed batches
```

**Review by awaelchli — APPROVED** [review]
great to have this clarified in the docs! 🎉 
I only have minor comments

File: docs/source/common/optimizers.rst [hunk 7]
```diff
@@ -130,7 +124,21 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
       scheduler2 = LambdaLR(optimizer2, ...)
       return [optimizer1, optimizer2], [scheduler1, scheduler2]
 
-   # Alternatively
+Schedulers in which ``.step()`` method is conditioned on metric values (for example the
+:class:`~torch.optim.lr_scheduler.ReduceLROnPlateau` scheduler) enforces that the output from ``configure_optimizers`` should be returned as dicts with the  keyword ``monitor`` defined as the
+metric that the scheduler should be conditioned on.
+
```
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply docs/source/common/optimizers.rst]: Done :]

File: docs/source/common/optimizers.rst [hunk 8]
```diff
@@ -141,15 +149,33 @@ Every optimizer you use can be paired with any `LearningRateScheduler <https://p
           {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
       )
 
-   # Same as above with additional params passed to the first scheduler
+By default, all schedulers will be called after each epoch ends. To change this behaviour, each scheduler should be
+returned as a dict which can contain the following keywords:
+
+* ``scheduler`` (required): the actual scheduler class
+* ``monitor`` (optional): same as above, defines the value that certain schedulers should be conditioned on
```
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply docs/source/common/optimizers.rst]: Done :]

**Review by Borda — APPROVED** [review]
